tools/clang/unittests/HLSL/ExtensionTest.cpp

```diff
@@ -1076,6 +1082,114 @@ TEST_F(ExtensionTest, SamplerExtensionIntrinsic) {
   CheckMsgs(disassembly.c_str(), disassembly.length(), expected, 1, true);
 }
 
+// Takes a string to match, and a regex pattern string, returns the first match at index [0],
+// as well as sub expressions starting at index [1]
+static std::vector<std::string> Match(const std::string &str, const std::string pattern) {
```

Comment:
It's minor, but I don't see the need to convert the output to a std::vector rather than keep it as a SmallVector

Reply:
I just didn't want the caller to always have to think about how big of a SmallVector to use. I could have used a SmallVectorImpl arg but that's kind of clumsy instead of having a return.